### ISSUE TITLE
Index data can only be added as XML comment, this makes JSON output invalid

### ISSUE THREAD

**Issue by willprice76** [SI4T/SI4T#4]
More and more implementations publish content in JSON format. The SI4T index data is added to the output as an XML comment, which is invalid JSON. This is not an issue if your deployer is correctly configured, as the SI4T storage extension will strip out the comment, but it would be nice if we didn't make the output invalid, just in case the storage extension is not working correctly.

**Comment by jhorsman** [comment]
A solution could be the extension data recently introduced in DD4T.

Using the extension data we can add the SI4T search indexing data in the DD4T data format. DD4T can still parse the JSON when the deployer does not remove the search indexing data from the extension data.

**Comment by RaimondKempees** [comment]
The extension data can only be used in the DD4T context, but not in other
implementations, so we can't really use that. We'll investigate the option
to transfer the data in a different format or in a different way.

**Comment by jhorsman** [comment]
@RaimondKempees Agree, the DD4T extension is a partial solution at best, not applicable for all environments.

**Comment by bkoopman** [comment]
And with the DD4T/DXA merge we will now also need to look at adding a solution to this for the DXA R2 data model. This can be done by creating a specific DXA R2 TBB that adds the index data in an extension node for DXA R2 JSON. The Deployer extension then needs to be smart enough to find the index data from several locations, or there are separate pluggable modules for it, based on what TBB is being used. I'll add this to the DXA roadmap so we can pick it up at some point.

**Comment by bkoopman** [comment]
Linked up to https://github.com/sdl/dxa-modules/issues/26 for now
